grid-demo/src/app/angular/advanced-row-selection/advanced-selection.directive.ts

```diff
@@ -0,0 +1,39 @@
+import { Directive, ElementRef, HostListener } from '@angular/core';
+import { ClrDatagrid } from '@clr/angular';
+
+const clrRowTagName = 'CLR-DG-ROW';
+
+@Directive({
+  selector: 'clr-datagrid[clrAdvancedSelection]',
+})
+export class AdvancedSelectionDirective {
```

Comment:
It would probably be good to change the cursor icon on hover to indicate that the row is clickable.

Also, this shift+click behavior is slightly different than Kendo, which takes the minimum or maximum currently selected index (depending on if the row you just clicked on is above or below) and selects everything inclusive. The behavior here doesn't seem to unselect anything.

Reply:
We could make the directive inject this style:

```css
clr-datagrid[clrAdvancedSelection] clr-dg-cell {
  cursor: pointer;
}
```

Reply:
I have this implemented. I can show you.

Reply:
1. The angular grid already supports the entire row being clickable and does not add a hover effect to the cursor. This is also consistent with the Kendo grid.

2. You're right that the shift+click behavior doesn't match Kendo's. I was using the shift+click behavior that Ivan implemented. I think this will be close enough and it will be hard to fight with Ivan's implementation. Do you think this is good enough? Do you think we should change Ivan's released implementation? Or do you think I should try to somehow override Ivan's shift+click behavior?

Reply:
I spaced and forgot that shift+click was implemented and this was just ctrl+click. I think this is fine.

I do still think the cursor to indicate click is better UX, but it's not hard for vsphere to add that themselves